src/routes/threlte/stores.ts

```diff
@@ -0,0 +1,125 @@
+import { Project as WasmProject, Realization as WasmRealization } from 'cadmium'
+import { writable } from 'svelte/store'
+import type { WorkBench, MessageHistory, Project, Realization, Entity, EntityType, SnapEntity, PointLikeById, PreviewGeometry } from "../../types"
+import { isArcEntity, isCircleEntity, isEntity, isFaceEntity, isLineEntity, isMeshFaceEntity, isPlaneEntity, isPoint3DEntity, isPointEntity } from "../../typeGuards"
+import { _isDevelopment } from "../+layout"
+
+// prettier-ignore
+const log = (function () { const context = "[stores.ts]"; const color = "hotpink"; return Function.prototype.bind.call(console.log, console, `%c${context}`, `font-weight:bold;color:${color};`) })()
+
+// @ts-ignore
+export const wasmProject = writable<WasmProject>({})
+export const project = writable<Project>(emptyProject())
+export const projectIsStale = writable(false)
+
+export const workbenchIndex = writable(0)
+export const workbench = writable<WorkBench>(emptyWorkBench())
+export const workbenchIsStale = writable(false)
+
+export const featureIndex = writable<number>(1000)
+export const extrusionFeatures = writable<Entity[]>([])
+export const wasmRealization = writable<WasmRealization>()
+export const realization = writable<Realization>(emptyRealization())
+export const realizationIsStale = writable(false)
+
+export const hiddenSketches = writable<string[]>([])
+export const sketchBeingEdited = writable("")
+export const sketchTool = writable("")
+
+// could be looking for 'face' or 'plane' or other things
+export const selectingFor = writable<EntityType[]>([])
+export const selectionMax = writable(1000)
+export const selectionMin = writable(0)
+
+export const currentlyMousedOver = writable<SnapEntity[]>([])
+export const currentlySelected = writable<Entity[]>([])
+export const snapPoints = writable<PointLikeById[]>([])
+export const previewGeometry = writable<PreviewGeometry[]>([])
+
+export const messageHistory = writable<MessageHistory[]>([])
+
+project.subscribe(store => log("[project]", store))
+workbenchIndex.subscribe(store => log("[workbenchIndex]", store))
+workbench.subscribe(store => log("[workbench]", store))
+workbenchIsStale.subscribe(store => log("[workbenchIsStale]", store))
+featureIndex.subscribe(store => log("[featureIndex]", store))
+extrusionFeatures.subscribe(store => log("[extrusionFeatures]", store))
```

Comment:
maybe all these log statements should be inside the `if (_isDevelopment()) {` block a few lines down?

Reply:
indeed! done in d7fb1bf03cdc9678489f6c55ece1958d2403dfc5